scripts/set-repo-secrets.sh

```
@@ -0,0 +1,130 @@
+  echo "Error: XML file not found at ${xml_file}."
+  exit 1
+fi
+
+# Update the certificate in the XML file using awk for better multiline support
+sudo awk -v cert="${certificate_content}" '
+BEGIN { cert_block = "<certificate>" cert "</certificate>" }
+/<certificate>.*<\/certificate>/ {
+    sub(/<certificate>.*<\/certificate>/, cert_block)
+}
+{ print }
+' "${xml_file}" > "${xml_file}.tmp" && sudo mv "${xml_file}.tmp" "${xml_file}"
```

Shellcheck (0.10.0)

[warning] sudo doesn't affect redirects. Use ..| sudo tee file

(SC2024)